### ISSUE TITLE
`houdini generate` misses `+layout.gql` queries

### ISSUE THREAD

**Issue by wlockiv** [HoudiniGraphql/houdini#1156]
### Describe the bug

# ALL OF THIS INFO AND MORE IS IN THE README OF THE REPRO REPO

## The Problem

The type `houdini generate` does not add `+layout.gql` query types to its children's generated `./.$houdini/PageData` types. This is causing my `svelte-check` command to fail in CI/CD pipelines.

I've only been able to reproduce this in `+layout.gql` files within `[param]` routes.

The types DO seem to get generated AFTER the development server has been started. 

### Severity

serious, but I can work around it

### Clone the repro & Install Deps

```sh
$ git clone https://github.com/wlockiv/houdini-layout-issue-repro.git
$ cd houdini-layout-issue-repro
$ pnpm i
```

### Simulate CI/CD pipeline command chain

I've made helper scripts called `clean` and `reproduce`:

```sh
# deletes the $houdini and .svelte-kit directories
$ pnpm run clean

#  basically runs my ci/cd pipeline commands in order
$ pnpm run reproduce
```

OR you can run them on your own:

```sh
# generate sveltekit types
$ pnpm svelte-kit sync

# generate houdini types
$ pnpm houdini generate

# run type checking on svelte files
$ pnpm svelte-check
```

### Note the error emitted from `svelte-check`

```sh
houdini-layout-issue-repro/src/routes/[id]/+page.svelte:6:11
Error: Property 'FilmLayoutQuery' does not exist on type '{ RootLayoutQuery: RootLayoutQueryStore; }'. (ts)

    $: ({ FilmLayoutQuery } = data);
</script>
```

### Reproduction

https://github.com/wlockiv/houdini-layout-issue-repro

**Comment by jycouet** [comment]
Hello,
First, thanks a lot for the reproduction, it helps to understand your issue.

_I almost didn't remember that we have a manual `houdini generate` cmd._
With the `houdini` vite plugin, it should take care of everything for you.

In dev mode you do: 
```bash
pnpm dev
pnpm svelte-check
```

In CI mode you do
```bash
pnpm build
pnpm svelte-check
```

I tried in your repo and seemed better than `pnpm run reproduce`.
Let us know.

--- 
Looking at the repo, I have a side note, I saw: 
```ts
{#if $RootLayoutQuery.data?.allFilms?.films}
    {#each $RootLayoutQuery.data.allFilms.films as film}
        <p><a href="/{film?.id}">{film?.title}</a></p>
    {/each}
{/if}
```
Maybe this can be nice as well
```ts
{#each $RootLayoutQuery.data?.allFilms?.films ?? [] as film}
  <p><a href="/{film?.id}">{film?.title}</a></p>
{:else}
  No films!
{/each}
```

**Comment by wlockiv** [comment]
> First, thanks a lot for the reproduction, it helps to understand your issue.

You're most welcome! It's the least I could do for soliciting free help!

I'll update my CI/CD pipeline and report back!

On the each/else blocks - is that behavior documented?? That's pretty cool but I can't find anything that describes what it's actually doing!

**Comment by jycouet** [comment]
Yeah, let us know 👍

---

https://svelte.dev/docs/logic-blocks#each

> An each block can also have an `{:else}` clause, which is rendered if the list is empty.
```ts
{#each todos as todo}
  <p>{todo.text}</p>
{:else}
  <p>No tasks today!</p>
{/each}
```
And I do the `?? []` to have an empty array if it's null somewhere.
I love it ^^

**Comment by wlockiv** [comment]
It actually worked really well! Thank you!!
